include/grpcpp/server_builder_impl.h

```diff
@@ -248,6 +265,16 @@ class ServerBuilder {
     ServerBuilder& RegisterCallbackGenericService(
         grpc::experimental::CallbackGenericService* service);
 
+    enum ExternalConnectionType {
+      CONNECTION_FROM_FD = 0  // in the form of a file descriptor
```

Comment:
Maybe...

Reply:
I would suggest renaming it to GRPC_HTTP2_FROM_FD since one of the possible ways that you'll see different connection types is if we decide to support different transports and thus different wire formats. The enum would indicate which wire format to use when interpreting the FD.

Reply:
I am not very sure about this since it is not clear to me how we choose other transports from the API level. I think now the other transports are linked with the credentials type. As a result, it seems like we do not need to specify the transport type again in the type here, e.g. a connection_from_fd type can be handled differently according to the credentials type linked.

Reply:
I'm ok with what you just said. But now that it's an "enum class", the CONNECTION prefix is redundant with the class scope. So it could just be FROM_FD so that the full specification is ExternalConnectionType::FROM_FD . I'll leave that decision to you and the eventual gRFC.